### PR TITLE
feat: Tailwind-based UI package

### PR DESCRIPTION
**Pull request by VanishMax**
Followed up by https://github.com/penumbra-zone/web/pull/1902

The Tailwind-based UI library.

Vite, in this case, builds the library code and produces simple JavaScript that takes React's `createElement` and applies the right classes to it. The styling responsibility for now is on the consumer's side. However, I exported one function for simple Tailwind configuration of consumer's `tailwind.config.ts`:

```ts
import { withPenumbra } from '@penumbra-zone/ui-tailwind/theme';

export default withPenumbra({
  content: [
    './app/**/*.{js,ts,jsx,tsx,mdx,css}',
    './src/**/*.{js,ts,jsx,tsx,mdx,css}',
  ],
});
```

Components list. Agenda: ✅ – implemented, 🟡 – not implemented (not needed for now), ❌ – i decided not to implement it.
1. ⚠️AccountSelector
2. ✅AddressViewComponent 
3. ✅AssetIcon 
4. ✅AssetSelector
5. ✅Button
6. ❌ButtonGroup
7. ✅Card
8. ❌CharacterTransition
9. ✅ConditionalWrap
10. ✅CopyToClipboardButton
11. ✅Density
12. ✅️Dialog
13. ✅Display
14. ✅DropdownMenu
15. 🟡FormField
16. ✅Grid
17. ✅Icon
18. ✅Identicon
19. ❌IsAnimatingProvider 
20. ✅MenuItem
21. ❌PenumbraUIProvider
22. ✅Pill
23. ✅Popover
24. ✅Progress
25. 🟡SegmentedControl
26. 🟡Slider
27. 🟡SwapInput
28. ✅Table
29. ✅Tabs
30. ✅Text
31. ✅TextInput
32. ✅Toast
33. 🟡Toggle
34. ✅Tooltip
35. 🟡ValueInput
36. ✅ValueView
37. ✅WalletBalance

## Future work

1. Improve `withPenumbra` hook to cover all config merging cases
2. Add tests to components, move old tests back and make them work
3. Export a `.css` file with all compiled Tailwind classes, so consumers wouldn't need Tailwind in their projects
4. Implement other UI components that were left undone for now

### CODE REVIEW

File: packages/ui-tailwind/src/Text/index.tsx [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+}: TextProps): string => {
+  const truncateClass = truncate ? 'truncate' : '';
+  const alignClass = align && ALIGN_OPTIONS.includes(align) ? `text-${align}` : '';
+  const decorationClass =
+    decoration && DECORATION_OPTIONS.includes(decoration) ? `line-${decoration}` : '';
+  const transformClass =
+    transform && TRANSFORM_OPTIONS.includes(transform) ? `text-${transform}` : '';
+  const breakClass = breakProp && BREAK_OPTIONS.includes(breakProp) ? `break-${breakProp}` : '';
+  const whitespaceClass =
+    whitespace && WHITESPACE_OPTIONS.includes(whitespace) ? `whitespace-${whitespace}` : '';
```
Comment (VanishMax) [inline packages/ui-tailwind/src/Text/index.tsx:93]: issue: dynamic classes are not recognized by the Tailwind's JIT compiler

File: packages/ui-tailwind/src/utils/color.ts [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+  } catch (_) {
+    throw new Error(`Color "${color}" not found in the registered Tailwind theme`);
+  }
+};
+
+/** This mapper class is needed to help Tailwind statically analyze the classes that could
+ * be produced from the `getThemeColorClass` function. */
+export const COLOR_CLASS_MAP: Record<ThemeColor, [string, string]> = {
+  'neutral.main': ['text-neutral-main', 'bg-neutral-main'],
+  'neutral.light': ['text-neutral-light', 'bg-neutral-light'],
```
Comment (VanishMax) [inline packages/ui-tailwind/src/utils/color.ts:37]: this color mapping is needed for Tailwind's compiler. It needs to see static classes, dynamic ones fail to be parsed

File: .changeset/wise-toes-bathe.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@penumbra-zone/ui': major
+---
```
Comment (grod220) [inline .changeset/wise-toes-bathe.md:3]: Ideally we can:
- rename our old package to `@penumbra-zone/ui-deprecated` and set it to private
- name the new package `@penumbra-zone/ui`, set it as the old version `12.4.0`, and keep this major bump for that. After changeset versions everything, it should publish the new stuff.

However, this probably requires changing all the import paths. Maybe good for a next PR?

File: packages/ui-tailwind/src/theme/tailwind-config.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    content: composeContent(config.content),
+    theme: {
+      ...config.theme,
+      extend: {
+        ...tailwindConfig.theme.extend,
+        ...config.theme?.extend,
+      },
+    },
+  };
+};
```
Comment (grod220) [inline packages/ui-tailwind/src/theme/tailwind-config.ts:67]: question: does this guarantee no conflicts with the consumers existing styles?

**Review by grod220 — APPROVED** [review]
What a heroic effort on this 🎊 . Huge win getting this fixed.

**Comment by VanishMax** [comment]
@grod220 do you have the idea why the tests are failing in the minifront? Seems like it is triggered by your changes

https://github.com/penumbra-zone/web/actions/runs/11724483746/job/32658624212?pr=1891

**Comment by grod220** [comment]
The "multiple copies of react" issue sounds like you are bundling react with the tailwind package. Think that means you need to make it a peerDep and not a main dependency.

File: .changeset/wise-toes-bathe.md (hunk 3 above)
Comment (VanishMax) [reply .changeset/wise-toes-bathe.md]: Did it here: https://github.com/penumbra-zone/web/pull/1902

**Pull request merged**